### DISCUSSION TITLE
Chat with PDF (going further)

### DISCUSSION THREAD

**Discussion by chriscousin** [FlowiseAI/Flowise#881] [Q&A]
Hello and thank you for this wonderful project. I have managed to create a chatbot that allows me to chat with a PDF document. However, I'm struggling to go further. For example, I would like the PDF document to become a contextual element, and be able to ask ChatGPT to write new sections that complement the existing PDF, while continuing to converse with ChatGPT about related topics. I'd like to be able to ask questions that go beyond the scope of the PDF alone. What advice do you have?

**Comment by chungyau97** [comment]
Hi @chriscousin,

You will be needing 2 flow to achieve your requirements

**To write new section**:
You have to use 3rd party platform [Zapier/Pipedream] to achieve this.
1. Flowise AI and User's message will be send to 3rd party.
2. 3rd party will then append into a public accessible file [Google Docs]
3. 3rd party convert file to PDF
4. 3rd party call Flowise `upsert document` flow through API

**Go beyond the scope of the PDF**: 
You will need to use an `Conversational Retrieval Agent` shown below:
![image](https://github.com/FlowiseAI/Flowise/assets/33013947/434b0e70-934f-4278-ab22-5ec2aa797fcc)

**Reply by chriscousin** [reply]
Thanks a lot @chungyau97 ! I will try to follow your advises. You rock !
